File: skills/tooling-java-generator/examples/LibraryExample.java
package com.example.utils;

import java.time.Instant;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

/**
 * Java Library Example: StringMetrics utility
 * Demonstrates: immutability, record types, collections, modern Java features
 */
public final class StringMetrics {

    public record Metrics(int length, int wordCount, Instant analyzed) {}

    private final List<String> history = new ArrayList<>();

    public Metrics analyze(String text) {
        if (text == null || text.isBlank()) {
            throw new IllegalArgumentException("Text cannot be null or blank");
        }

        history.add(text);
        int wordCount = text.split("\\s+").length;

        return new Metrics(text.length(), wordCount, Instant.now());
    }

    public List<String> getHistory() {
        return Collections.unmodifiableList(history);
    }
}

File: skills/tooling-csharp-generator/examples/LibraryExample.cs
namespace Example.Utils;

/// <summary>
/// C# Library Example: TextAnalyzer utility
/// Demonstrates: records, nullable reference types, LINQ, modern C# features
/// </summary>
public sealed class TextAnalyzer
{
    public record AnalysisResult(int Length, int WordCount, DateTime Analyzed);

    private readonly List<string> _history = new();

    public AnalysisResult Analyze(string text)
    {
        ArgumentException.ThrowIfNullOrWhiteSpace(text);

        _history.Add(text);
        var wordCount = text.Split(' ', StringSplitOptions.RemoveEmptyEntries).Length;

        return new AnalysisResult(text.Length, wordCount, DateTime.UtcNow);
    }

    public IReadOnlyList<string> GetHistory() => _history.AsReadOnly();

    public void Clear() => _history.Clear();
}
